operator/pkg/functional/compare.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package functional
+
+func CompareMaps[T, U comparable](a, b map[T]U) bool {
+	if len(a) != len(b) {
+		return false
+	}
+	for key := range a {
+		if a[key] != b[key] {
+			return false
+		}
+	}
+	return true
+}
+
+func CompareMapsFn[T comparable, U any](a, b map[T]U, fn func(U, U) bool) bool {
```

Comment:
Same-ish question here, `maps.EqualFunc` does _nearly_ the same thing but with checks on key presence.